update_secret.go

```diff
@@ -32,38 +40,75 @@ func main() {
 
 	cp := flag.String("cert-path", "./build/secrets/server_crt.pem", "The path to your end-entity certificate")
 	kp := flag.String("key-path", "./build/secrets/server_key.pem", "The path to your end-entity private key")
-    flag.Parse()
+  flag.Parse()
 
-	cert, err := os.ReadFile(*cp)
+	cpem, err := os.ReadFile(*cp)
 	if err != nil {
-    	panic("failed to read certificate: " + err.Error())
-    } 
+    	log.Fatalf("failed to read certificate: %v\n", err.Error())
+  } 
 
-	key, err := os.ReadFile(*kp)
+	kpem, err := os.ReadFile(*kp)
 	if err != nil {
-    	panic("failed to parse private key: " + err.Error())
-    }
+			log.Fatalf("failed to parse private key: %v\n", err.Error())
+	}
+
+	cblock, _ := pem.Decode(cpem)
+	if cblock == nil || cblock.Type != "CERTIFICATE" {
+		log.Fatal("failed to decode PEM block containing certificate")
+	}
+
+	kblock, _ := pem.Decode(kpem)
+	if kblock == nil || kblock.Type != "PRIVATE KEY" {
+		log.Fatal("failed to decode PEM block containing private key")
+	}
 
 	sc := getk8sSecretClient()
+	cert := string(cpem)
+	key := string(kpem)
 
 	// Retry updating secret until you no longer get a conflict error. 
 	// This way, you can preserve changes made by other clients between.
 	// Ref: https://github.com/kubernetes/client-go/blob/master/examples/create-update-delete-deployment/main.go
 	retryErr := retry.RetryOnConflict(retry.DefaultRetry, func() error {
 		s, getErr := sc.Get(context.TODO(), "tls-secret", metaV1.GetOptions{})
-		if getErr != nil {
-			panic(fmt.Errorf("Failed to get latest version of secret tls-secret: %v", getErr))
+
+		// If secret "tls-secret" is not found
+		if errors.IsNotFound(getErr) {
+
+			sd := make(map[string]string)
+			sd["tls.crt"] = cert
+			sd["tls.key"] = key
+
+			s_ := &v1.Secret{
+				Type: "kubernetes.io/tls",
+				ObjectMeta: metaV1.ObjectMeta{
+					Name: "tls-secret",
+					Namespace: "default",
+				},
+				StringData: sd,
+			}
+
+			_, createErr := sc.Create(context.TODO(), s_, metaV1.CreateOptions{})
+			if createErr != nil {
+				log.Fatalf("Update failed: %v\n", createErr)
+			}
+			return createErr
 		}
-		s.Data["tls.crt"] = []byte(cert)
-		s.Data["tls.key"] = []byte(key)
+
+		// If you forget to add this statement, you will get an error (panic: assignment to entry in nil map)
+		if s.StringData == nil {
+			s.StringData = map[string]string{}
+		}
+
+		s.StringData["tls.crt"] = cert
+		s.StringData["tls.key"] = key
 
 		_, updateErr := sc.Update(context.TODO(), s, metaV1.UpdateOptions{})
 		return updateErr
 	})
 
 	if retryErr != nil {
-		panic(fmt.Errorf("Update failed: %v", retryErr))
+		log.Fatalf("Update failed: %v\n", retryErr)
 	}
-	fmt.Println("Secret tls-secret is successfully updated")
-
-}
\ No newline at end of file
+	log.Fatal("Secret tls-secret is successfully updated")
```

Comment:
`log.Fatal` returns exit code 1. It recommended to return exit code 0 if it finished successfully.